jdbc/src/main/java/com/yahoo/ycsb/db/JdbcDBClient.java

```diff
@@ -474,7 +486,21 @@ public Status insert(String tableName, String key, HashMap<String, ByteIterator>
       for (String value: fieldInfo.getFieldValues()) {
         insertStatement.setString(index++, value);
       }
-      int result = insertStatement.executeUpdate();
+      int result;
+      if (batchSize > 0) {
+        insertStatement.addBatch();
+        if (++numRowsInBatch % batchSize == 0) {
+          int[] results = insertStatement.executeBatch();
+          for (int r : results) {
+            if (r != 1) {
+              return Status.ERROR;
+            }
+          }
+        }
+        return Status.OK;
```

Comment:
You might consider taking a look at the Status codes for Batch inserts in Mongo: https://github.com/brianfrankcooper/YCSB/blob/master/mongodb/src/main/java/com/yahoo/ycsb/db/MongoDbClient.java.

Essentially, when an operation is added to the batch you return Status.BATCHED_OK, when you submit the whole batch then you return Status.OK. This way users can differentiate between the two actions when looking at the results.